### PR TITLE
Adds a medkit wall cabinet and makes empty medkits printable.

### PR DESCRIPTION
**Pull request by shellspeed1**
## About The Pull Request
This pr adds a new medkit wall cabinet to the game that works the exact same way as a fire extinguisher cabinets . It also enables medbay to print empty medkits to put in them as well as to hold supplies for doctors. Both are unlocked with biological technology researched and are printable at the medical protolathe. If placed by mappers it will contain a basic medkit. Also thanks to BrokenOculus for making sprites. 

## Why It's Good For The Game
This PR is meant to encourage medbay to distribute supplies around the station to make it easier to handle emergencies and give them something to do to help make medbay a little less reactive and a bit more proactive. Also helps to make supplies a bit more visible since people have a tendency to fail to look inside the chem fridge for medication the chemist has made.

## Changelog
:cl:
add: Adds medkit wall cabinets and empty medkits to the medical protolathe.
add: Medkit wall cabinents and empty medkits are now part of biological technology.
/:cl:

### CODE REVIEW

**Comment by ExcessiveUseOfCobblestone** [comment]
all /obj's should be supported for going into a cabinet, it should just filter per type what is allowed to go  in

**Comment by shellspeed1** [comment]
> 
> 
> all /obj's should be supported for going into a cabinet, it should just filter per type what is allowed to go in

Not sure what you mean by that, would you mind explaining in a bit more detail?

**Comment by Time-Green** [comment]
> > all /obj's should be supported for going into a cabinet, it should just filter per type what is allowed to go in
> 
> Not sure what you mean by that, would you mind explaining in a bit more detail?

He means "please add support for every item to be wall cabineted easily without having to completely rewrite everything"

**Comment by ExcessiveUseOfCobblestone** [comment]
if I have to make a new var then rewrite the entire proc just to support particular objects (when just typing the var as /obj/ won't break anything) that's really yikes.

If I wanted a cabinet that added clownhorns all I should have to do is subtype then set the "what can go into this cabinet" var to clownhorns without having to do anything else.

**Comment by shellspeed1** [comment]
I'll see what I can do.

**Comment by shellspeed1** [comment]
I managed to make allowed item a var, however having issues making a variable to set what item it spawns with.

**Comment by cacogen** [comment]
We used to have medical wall cabinets with shit in them that was moved to closets interestingly enough

**Comment by shellspeed1** [comment]
I was wondering why we didn't have them.

File: code/game/objects/structures/extinguisher.dm [hunk 1]
```diff
@@ -9,7 +9,11 @@
 	integrity_failure = 50
 	var/obj/item/extinguisher/stored_extinguisher
 	var/opened = FALSE
+	//set allowed item to determine what item is allowed inside.
+	var/alloweditem
+	alloweditem = /obj/item/extinguisher
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/objects/structures/extinguisher.dm]: bruhminium

Comment (ExcessiveUseOfCobblestone) [reply code/game/objects/structures/extinguisher.dm]:
```DM
var/alloweditem = /obj/item/extinguisher
```

File: code/game/objects/structures/extinguisher.dm [hunk 2]
```diff
@@ -9,7 +9,11 @@
 	integrity_failure = 50
 	var/obj/item/extinguisher/stored_extinguisher
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/objects/structures/extinguisher.dm]:
```DM
var/obj/store_object
```

you'll need to rename all `stored_extinguisher` to `stored_object` tho

Comment (shellspeed1) [reply code/game/objects/structures/extinguisher.dm]: why though?

Comment (shellspeed1) [reply code/game/objects/structures/extinguisher.dm]: eh whatever, did it.

File: code/game/objects/structures/extinguisher.dm (hunk 1 above)
Comment (shellspeed1) [reply code/game/objects/structures/extinguisher.dm]: aaaaaaaaaaa, fixed it.

**Comment by Poojawa** [comment]
I think this would benefit further by having a generic cabinet as a parent/basic frame. which would then turn into whichever applicable item is attached to it.

then swap all the paths from `/obj/structure/extinguisher_cabinet` to being `/obj/structure/item_cabinet`  

with `/obj/structure/item_cabinet/fire_extinguisher` , `/obj/structure/item_cabinet/medical` , `/obj/structure/item_cabinet/gun_rack`  etc etc

**Comment by shellspeed1** [comment]
> 
> 
> I think this would benefit further by having a generic cabinet as a parent/basic frame. which would then turn into whichever applicable item is attached to it.
> 
> then swap all the paths from `/obj/structure/extinguisher_cabinet` to being `/obj/structure/item_cabinet`
> 
> with `/obj/structure/item_cabinet/fire_extinguisher` , `/obj/structure/item_cabinet/medical` , `/obj/structure/item_cabinet/gun_rack` etc etc
> 
> you could then make defib racks attach in with this too.

I am unsure, partially because defibs wall mounts are an actual machine.

**Comment by Poojawa** [comment]
Yee, Technically out of scope as well, but the thought occurred so

I believe they're only machinery so that they receive the effects of APC charging.

File: code/game/objects/structures/extinguisher.dm [hunk 3]
```diff
@@ -142,13 +146,56 @@
...
+	. = ..()
+	if(building)
+		setDir(ndir)
+		pixel_x = (dir & 3)? 0 : (dir == 4 ? -27 : 27)
+		pixel_y = (dir & 3)? (dir ==1 ? -30 : 30) : 0
+		opened = TRUE
+		icon_state = "medkit_empty"
+	else
+		//set this for item spawned inside
+		stored_object = new /obj/item/storage/firstaid/regular/(src)
```
Comment (nemvar) [inline code/game/objects/structures/extinguisher.dm]: Every medkit cabinet that gets spawned without building == TRUE has both a first aid kit AND an extinguisher in it. pls fix.

File: code/game/objects/structures/extinguisher.dm [hunk 4]
```diff
@@ -142,13 +146,56 @@
...
+
+//sprite code
+/obj/structure/extinguisher_cabinet/medkit/update_icon()
+	if(!opened)
+		icon_state = "medkit_closed"
+		return
+	if(stored_object)
+		if(istype(stored_object, /obj/item/storage/firstaid/regular)) //leave this in case someone decides to add more sprites for different kits
```
Comment (nemvar) [inline code/game/objects/structures/extinguisher.dm]: If someone decides to add more sprites to this, they can just add this code themselves. I would remove it.

File: code/game/objects/structures/extinguisher.dm [hunk 5]
```diff
@@ -142,13 +146,56 @@
...
+	//sets allowed item for the cabinet. Setting it to main type will allow subtypes.
+	alloweditem = /obj/item/storage/firstaid
+
+//mapcode
+/obj/structure/extinguisher_cabinet/medkit/Initialize(mapload, ndir, building)
+	. = ..()
+	if(building)
+		setDir(ndir)
+		pixel_x = (dir & 3)? 0 : (dir == 4 ? -27 : 27)
+		pixel_y = (dir & 3)? (dir ==1 ? -30 : 30) : 0
```
Comment (nemvar) [inline code/game/objects/structures/extinguisher.dm]: Make the pixel_x/pixel_y thing variables on the item and get rid of this entire duplicate code block.

File: code/game/objects/structures/extinguisher.dm (hunk 3 above)
Comment (shellspeed1) [reply code/game/objects/structures/extinguisher.dm]: I'm not sure how your getting that issue as on my end it isn't happening.

File: code/game/objects/structures/extinguisher.dm (hunk 5 above)
Comment (shellspeed1) [reply code/game/objects/structures/extinguisher.dm]: No longer applies. Anyway that just sets which side of the wall its on if I understand.

File: code/game/objects/structures/extinguisher.dm (hunk 4 above)
Comment (shellspeed1) [reply code/game/objects/structures/extinguisher.dm]: made sprites vars so this no longer applies.

File: code/game/objects/structures/extinguisher.dm (hunk 3 above)
Comment (shellspeed1) [reply code/game/objects/structures/extinguisher.dm]: If you can confirm that this is an issue, I'll take a look at it. After testing it again on my side still wasn't having issues.

File: code/game/objects/structures/extinguisher.dm [hunk 6]
```diff
@@ -7,41 +7,53 @@
 	density = FALSE
 	max_integrity = 200
 	integrity_failure = 50
-	var/obj/item/extinguisher/stored_extinguisher
+	var/obj/item/extinguisher/stored_object
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/objects/structures/extinguisher.dm]:
```suggestion
	var/obj/item/stored_object
```

**Comment by shellspeed1** [comment]
Fuck, missed that one.

**Comment by shellspeed1** [comment]
Meh, I might redo this later. Seems like people would like to limit the availability of medical supplies for right now anyway.

**Pull request closed without merging**